### PR TITLE
Implement numpy(force=True)

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109636
* #109634

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_dynamo/variables/tensor.py [hunk 1]
```diff
@@ -488,16 +488,15 @@ def make_const_size_variable(x, **options):
...
-                "call_function",
-                torch.detach,
-                *proxy_args_kwargs([self], {}),
+                "call_method", "detach", *proxy_args_kwargs([self], {})
             )
+            if "force" in kwargs and kwargs["force"].as_python_constant():
+                # TODO Add resolve_conj and resolve_neg once we support complex tensors
+                proxy = tx.output.create_proxy(
+                    "call_method", "cpu", *proxy_args_kwargs([self], {})
+                )
```
Comment (ezyang) [inline torch/_dynamo/variables/tensor.py:498]: I was wondering if it is correct to unconditionally call cpu but I guess the answer is that it is.

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/variables/tensor.py (hunk 1 above)
Comment (lezcano) [reply torch/_dynamo/variables/tensor.py:498]: As per the docs:
>If force is True this is equivalent to calling t.detach().cpu().resolve_conj().resolve_neg().numpy()
